index.js

```diff
@@ -568,11 +594,16 @@ exports.hook_rcpt = function (next, connection, params) {
     runCheck();
 };
 
-exports.real_rcpt_handler = function (next, connection, params) {
+exports.real_rcpt_handler = async function (next, connection, params) {
```

Comment:
Mixing async with callbacks is dangerous. It should be either one or another, but not both. If it is async, the method should take only `connection` and `params` as the arguments, and not `next` as returning the function should replace using the callback.

Reply:
updated accordlingly